### ISSUE TITLE
overridesdir detection is unreliable

### ISSUE THREAD

**Issue by jtojnar** [fedora-modularity/libmodulemd#466]
When trying to build without PyGObject installed (in a build chroot), the build fails with:

```
bindings/python/meson.build:36:2: ERROR: Problem encountered: Failed to determine Python 3 pygobject overridedir
```

It turns out the detection script tries to load the `gi` directory from the source tree, as evidenced by the following patch:

```patch
--- a/bindings/python/meson.build
+++ b/bindings/python/meson.build
@@ -4,24 +4,33 @@ import sysconfig
 
 libdir = sysconfig.get_config_var('SCRIPTDIR')
 
+print('libdir', libdir)
+
 if not libdir:
   libdir = '/usr/lib'
 
 try:
   import gi
+  print('dir', dir(gi))
+  for p in dir(gi): print(p, getattr(gi, p))
   overridedir = gi._overridesdir
+  print('try', overridedir)
 except ImportError:
   purelibdir = sysconfig.get_path('purelib')
   overridedir = os.path.join(purelibdir, 'gi', 'overrides')
+  print('except', overridedir)
 
 if overridedir.startswith(libdir): # Should always be True..
   overridedir = overridedir[len(libdir) + 1:]
+  print('if', overridedir)
 
 print(overridedir)
 '''
 
 # Python 3
 ret = run_command([python3, '-c', get_overridedir])
+message(ret.stdout())
+message(ret.stderr())
 
 if ret.returncode() != 0
   error('Failed to determine Python 3 pygobject overridedir')
```

```
Message: libdir /nix/store/vs4vj1yzqj1bkcqkf3b6sxm6jfy1gb4j-python3-3.7.7/lib
dir ['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
__doc__ None
__file__ None
__loader__ <_frozen_importlib_external._NamespaceLoader object at 0x7ffff754b150>
__name__ gi
__package__ gi
__path__ _NamespacePath(['/build/source/bindings/python/gi'])
__spec__ ModuleSpec(name='gi', loader=<_frozen_importlib_external._NamespaceLoader object at 0x7ffff754b150>, submodule_search_locations=_NamespacePath(['/build/source/bindings/python/gi']))

Message: Traceback (most recent call last):
  File "<string>", line 16, in <module>
AttributeError: module 'gi' has no attribute '_overridesdir'

```

**Comment by jtojnar** [comment]
With pygobject3 installed, I see:

```
Message: libdir /nix/store/vs4vj1yzqj1bkcqkf3b6sxm6jfy1gb4j-python3-3.7.7/lib
…
try /nix/store/g5142vy7c74iccsy7gp3qxnqpg0c7nlb-python3.7-pygobject-3.36.0/lib/python3.7/site-packages/gi/overrides
/nix/store/g5142vy7c74iccsy7gp3qxnqpg0c7nlb-python3.7-pygobject-3.36.0/lib/python3.7/site-packages/gi/overrides
```

which is not ideal either, since we want to install this to our own prefix (e.g. `/nix/store/50pq7c1wsblc94xfk1x2mp4b2gig9pnp-libmodulemd-2.9.2/lib/python3.7/site-packages/gi/overrides`).

**Comment by sgallagher** [comment]
I merged PR #469, so this should be fixed.

@jtojnar Would you be willing to contribute a test for NixOS build and install similar to those we have for Mageia, OpenMandriva and OpenSUSE? See the [`.travis`](https://github.com/fedora-modularity/libmodulemd/tree/master/.travis) subdirectory in the checkout for details.

**Comment by jtojnar** [comment]
Looking at it more, I am not even sure if we should keep libmodulemd in our repositories. It was only added as a dependency for libdnf used by rpm-ostree but its maintainer abandoned `rpm-ostree` and I have no idea what it could be used for on NixOS.
